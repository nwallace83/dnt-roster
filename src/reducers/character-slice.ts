import { createSlice } from '@reduxjs/toolkit'
import Character, { getNewCharacter } from '../types/character'
interface CharacterState extends Character {
}
const initialState: CharacterState =  getNewCharacter()

export const characterSlice = createSlice({
    name:'character',
    initialState: initialState,
    reducers: {
        saveCharacter: (state,payload: {type: string, payload: Character}) => {
            const character = payload.payload
            return character
        },
        clearCharacter: () => {
            return initialState
        },
        setCharacterField: (state,payload: {type: string, payload: {field: string, value: string | boolean | number}}) => {
            return {...state,[payload.payload.field]:payload.payload.value}
        },
        setCharacrCraftingField: (state,payload: {type: string, payload: {field: string, value: string | boolean | number}}) => {
          return {...state,[payload.payload.field]:payload.payload.value}
      },
    }
})
export const { clearCharacter, saveCharacter }  = characterSlice.actions
export default characterSlice.reducer