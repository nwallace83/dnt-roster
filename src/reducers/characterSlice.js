import { createSlice } from '@reduxjs/toolkit'

export const characterSlice = createSlice({
    name:'character',
    initialState: {
        characterName: "",
        primaryWeapon1: "",
        primaryWeapon2: "",
        primaryRole: "",
        primaryArmor: "",
        primaryGS: "",
        secondaryWeapon1: "",
        secondaryWeapon2: "",
        secondaryRole: "",
        secondaryArmor: "",
        secondaryGS: "",
        discordUserName: ""
    },
    reducers: {
        saveCharacter: (state,character) => {
            return character.payload
        },
        clearCharacter: (state) => {
            return {
                characterName: "",
                primaryWeapon1: "",
                primaryWeapon2: "",
                primaryRole: "",
                primaryArmor: "",
                primaryGS: "",
                secondaryWeapon1: "",
                secondaryWeapon2: "",
                secondaryRole: "",
                secondaryArmor: "",
                secondaryGS: "",
                discordUserName: ""
            }
        }
    }
})

export const { clearCharacter, saveCharacter }  = characterSlice.actions

export default characterSlice.reducer