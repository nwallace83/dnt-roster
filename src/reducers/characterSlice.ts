import { createSlice } from '@reduxjs/toolkit'
import { current } from 'immer'
import Character, { CharacterCrafting } from '../interfaces/character'

interface State {
    characterName: string,
    primaryWeapon1: string,
    primaryWeapon2: string,
    primaryRole: string,
    primaryArmor: string,
    primaryGS: string,
    secondaryWeapon1: string,
    secondaryWeapon2: string,
    secondaryRole: string,
    secondaryArmor: string,
    secondaryGS: string,
    discordUserName: string,
    inactive: boolean,
    crafting: {
            weaponSmithing: boolean,
            armoring: boolean,
            engineering: boolean,
            jewelCrafting: boolean,
            arcana: boolean,
            cooking: boolean,
            furnishing: boolean,
    } 
}
const initialState: State =  {
        characterName: '',
        primaryWeapon1: '',
        primaryWeapon2: '',
        primaryRole: '',
        primaryArmor: '',
        primaryGS: '',
        secondaryWeapon1: '',
        secondaryWeapon2: '',
        secondaryRole: '',
        secondaryArmor: '',
        secondaryGS: '',
        discordUserName: '',
        inactive: false,
        crafting: {
                weaponSmithing: false,
                armoring: false,
                engineering: false,
                jewelCrafting: false,
                arcana: false,
                cooking: false,
                furnishing: false
        }
    }

export const characterSlice = createSlice({
    name:'character',
    initialState: initialState,
    reducers: {
        saveCharacter: (state,payload: {type: string, payload: Character}) => {
            const character = payload.payload
            if (character && character.characterName && character.crafting) {
                return character
            } else {
                return {...state,initialState}
            }
        },
        clearCharacter: () => {
            return initialState
        },
        toggleTradeSkill: (state,payload: {type: string, payload: string}) => {
            const tradeSkill = payload.payload
            let currentState: State = current(state)
            let newValue: boolean = !currentState.crafting[tradeSkill as keyof CharacterCrafting]
            return {...state,crafting: {...state.crafting,[tradeSkill]: newValue}}
        }
    }
})

export const { clearCharacter, saveCharacter, toggleTradeSkill }  = characterSlice.actions

export default characterSlice.reducer