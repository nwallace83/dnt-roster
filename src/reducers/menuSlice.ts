import { createSlice } from '@reduxjs/toolkit'

interface State {
    activeTab: string | null
}

const initialState: State = {activeTab: localStorage.getItem('activeTab') ? localStorage.getItem('activeTab') : 'roster'}

export const menuSlice = createSlice({
    name:'menu',
    initialState: initialState,
    reducers: {
        changeTab: (state,tab: {type: string, payload: string}): State => {
            localStorage.setItem('activeTab',tab.payload)
            return {...state,activeTab: tab.payload}
        }
    }
})

export const { changeTab }  = menuSlice.actions

export default menuSlice.reducer