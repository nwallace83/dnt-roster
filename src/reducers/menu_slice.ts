import { createSlice } from '@reduxjs/toolkit'

interface MenuState {
    activeTab: string
}

const initialState: MenuState = {activeTab: localStorage.getItem('activeTab') ?? 'roster'}

export const menuSlice = createSlice({
    name:'menu',
    initialState: initialState,
    reducers: {
        changeTab: (state,tab: {type: string, payload: string}): MenuState => {
            localStorage.setItem('activeTab',tab.payload)
            return {...state,activeTab: tab.payload}
        }
    }
})

export const { changeTab }  = menuSlice.actions

export default menuSlice.reducer