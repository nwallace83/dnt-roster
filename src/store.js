import { configureStore } from '@reduxjs/toolkit'
import menuReducer from './reducers/menuSlice'
import sessionReducer from './reducers/sessionSlice'
import characterReducer from './reducers/characterSlice'
import rosterReducer from './reducers/rosterSlice'

export default configureStore({
    reducer: {
        menu: menuReducer,
        session: sessionReducer,
        character: characterReducer,
        roster: rosterReducer
    }
})