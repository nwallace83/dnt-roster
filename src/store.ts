import { configureStore } from '@reduxjs/toolkit'
import menuReducer from './reducers/menu-slice'
import sessionReducer from './reducers/session-slice'
import rosterReducer from './reducers/roster-slice'
import characterReducer from './reducers/character-slice'
import { reducer as toastrReducer } from 'react-redux-toastr'

let store =  configureStore({
    reducer: {
        menu: menuReducer,
        session: sessionReducer,
        roster: rosterReducer,
        character: characterReducer,
        toastr: toastrReducer
    }
})

export default store

export type RootState = ReturnType<typeof store.getState>
export type AppDispatch = typeof store.dispatch