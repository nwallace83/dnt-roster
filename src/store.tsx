import { configureStore } from '@reduxjs/toolkit'
import menuReducer from './reducers/menuSlice'
import sessionReducer from './reducers/sessionSlice'
import rosterReducer from './reducers/rosterSlice'
import { reducer as toastrReducer } from 'react-redux-toastr'

let store =  configureStore({
    reducer: {
        menu: menuReducer,
        session: sessionReducer,
        roster: rosterReducer,
        toastr: toastrReducer
    }
})

export default store

export type RootState = ReturnType<typeof store.getState>
export type AppDispatch = typeof store.dispatch