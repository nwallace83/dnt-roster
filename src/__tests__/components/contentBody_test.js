import {render, screen} from '@testing-library/react'
import '@testing-library/jest-dom'
import ContentBody from '../../components/contentBody'
import { Provider } from 'react-redux'
import store from '../../store'
import { changeTab } from '../../reducers/menuSlice'
import renderer from 'react-test-renderer';
import { act } from 'react-dom/test-utils'

jest.mock("../../components/editCharacter", () => () => {  
        return <mock-editCharacter data-testid="edit-character" />
    }
)

jest.mock("../../components/crafters", () => () => {  
        return <mock-crafters data-testid="crafters" />
    }
)

jest.mock("../../components/roster", () => () => {  
        return <mock-roster data-testid="roster" />
    }
)

describe('ContentBody',() => {
    let container

    beforeEach(() => {
        render(<Provider store={store}><ContentBody /></Provider>)
        container = renderer.create(<Provider store={store}><ContentBody /></Provider>)
    })

    test('editCharacter is rendered', () => {
        act(() => {
            store.dispatch(changeTab('editCharacter'))
        })
        expect(screen.getByTestId('edit-character')).toBeInTheDocument()
        expect(container.toJSON()).toMatchSnapshot()
    })
    
    test('crafters is rendered', () => {
        act(() => {
            store.dispatch(changeTab('crafters'))
        })
        expect(screen.getByTestId('crafters')).toBeInTheDocument()
        expect(container.toJSON()).toMatchSnapshot()
    })
    
    test('roster is rendered', () => {
        act(() => {
            store.dispatch(changeTab(''))
        })
        expect(screen.getByTestId('roster')).toBeInTheDocument()
        expect(container.toJSON()).toMatchSnapshot()
    })

})
