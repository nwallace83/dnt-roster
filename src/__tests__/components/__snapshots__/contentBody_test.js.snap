// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentBody crafters is rendered 1`] = `
<mock-crafters
  data-testid="crafters"
/>
`;

exports[`ContentBody editCharacter is rendered 1`] = `
<mock-editCharacter
  data-testid="edit-character"
/>
`;

exports[`ContentBody roster is rendered 1`] = `
<mock-roster
  data-testid="roster"
/>
`;
